frontend/src/metabase/query_builder/components/template_tags/TagEditorSidebar.jsx

```diff
@@ -50,6 +55,12 @@ export default class TagEditorSidebar extends React.Component {
       setParameterValue,
       onClose,
     } = this.props;
+
+    // Sync card's parameters with the template tags;
+    card.parameters = getTemplateTagParameters(
```

Comment:
While this seems to be a bit gross, it's the only compromise that makes sense, until we can refactor `TagEditorSidebar` into a full-blown `ParameterEditingSidebar` (and that's also when it's worth pursuing it).

Reply:
Would it be possible to stuff this into a query builder action or reducer? It'd be nice to avoid mutating state in a render method in case we never get around to adding `ParameterEditingSidebar`